.github/workflows/common.sh

```diff
@@ -67,6 +67,11 @@ function prepare_git_repo() {
 function check_remote_branch() {
   local target_branch="${1}"
 
+  # It is needed to first fetch remote branches before checking for remote
```

Comment:
You could use `git -C "${SDK_OUTER_TOPDIR}" ls-remote --refs --heads --exit-code origin "${target_branch}" >/dev/null`, so this function could be:

```bash
function check_remote_branch() {
    local target_branch="${1}"
    # this makes "set +e" below to be effective only within the function scope
    local -
    set +e
    git -C "${SDK_OUTER_TOPDIR}" ls-remote --refs --heads --exit-code origin "${target_branch}" >/dev/null
}
```

Reply:
Yes, that could work as well.
In that case, however, we have to update all call sites of `check_remote_branch` like `if check_remote_branch branch_name; then error`.
Is that what you want?

Reply:
Oh. Right. Too many reverses just ended up confusing me. I'd avoid the rewrites, so:

```bash
# Returns zero if the passed branch was not found on the remote server, otherwise returns non-zero.
function check_remote_branch() {
    local target_branch="${1}"
    if git -C "${SDK_OUTER_TOPDIR}" ls-remote --refs --heads --exit-code origin "${target_branch}" >/dev/null; then
        return 1
    fi
    return 0
}
```

Reply:
Done. Thanks